Add GS Platform Access Routes documentation (#317)
giantswarm/docs: src/content/basics/giant-swarm-platform-access/index.md

+++
title = "The Giant Swarm Platform Access Routes"
description = "General Giant Swarm Platform Access Routes Overview"
date = "2019-05-08"
weight = 20
type = "page"
categories = ["basics"]
+++

# Giant Swarm Platform Access Routes

## Intro

> Access to Giant Swarm platforms can be split into two parts. 

Reviewer: I find the `platforms` term not specific enough. This should be specified and aligned with the title. See my other comment.

> 1. Admin Access - designated for the Giant Swarm employees for managing/development/support purposes.

Reviewer: ```suggestion
1. Admin Access - designated for Giant Swarm staff for managing/development/support purposes.
```

> 2. User Access - designated for GiantSwarm customers to interact with the offered services.

Reviewer: ```suggestion
2. User Access - designated for Giant Swarm customers to interact with the offered services.
```

> Both are managed via Giant Swarm VPN hopp that enables connection through jumphost of the cluster.

Reviewer 2: What is a `hopp `?

Reviewer: My understand so far is that VPN protection of the Giant Swarm API is only optional. We have some installations where the API is reachable from everywhere.

Reviewer 3: Yes,  GS API protection is optional

Reviewer: Anyway, I would try to find a more suitable term. Jump host? Bastion Host? VPN entry point? Server?

> Both are managed via Giant Swarm VPN hopp that enables connection through jumphost of the cluster.

Reviewer: ```suggestion
Both are managed via Giant Swarm Virtual Private Network (VPN) hopp that enables connection through jumphost of the cluster.
```

When introducing an abbreviation/acronym, we use the long form and set the short form in brackets. After that, the short form can be used.

Giant Swarm uses two different providers of VPN to improve the resilience of offered services.

## General VPN Connection Schema

> ![](/img/site-to-site-vpn.png)

Reviewer: It should be possible to place the image in the same folder as `index.md`. Can you please try that? It would simplify previews and also management, as it's more clear where an image is used.

## Admin Access

VPN access is managed via certificates (public/private keys) and secured connection between both VPNs.
Certificates management is handled by one of GS private components. 

> The general principle is that certificate is configured for each individual Giant Swarm employee.

Reviewer: ```suggestion
The general principle is that certificate is configured for each individual Giant Swarm staff member.
```

VPN Secured access points:

* SSH - SSH Admin access is based on the Github SSO. Only users residing in the Github GS Organization are allowed to authenticate.  
  Following process describes more in detail the procedure of SSH authentication:
  ![](/img/ssh_access_process.png)

>   SSH access to CP allows Giant Swarm also to manage and connect to underlying Data Platforms of the Customer.

Reviewer 2: I would explain what is `CP`, so people not familiarized with our platform can understand it

Reviewer: +1. As above, please use the long form when first introducing the acronym.

* CP Kubernetes API - Usage of Kuberentes API on CP also follows the authentication principles of the SSH connection. 
The Github SSO based on GS Organization is used to grant access. 

***Cloud Provider Access***

> At the current stage internal GS operator components responsible for managing clusters lifecycle at Customer's Cloud 

Reviewer 2: I would rather not use acronyms

Provider are granted the admin rights by the Customer to the given API.
This point is under discussion and might be limited in the near future to limited permissions allowing management of the clusters only.

> Operator secret used for authentication with Cloud Provider is stored in k8s api. 

Reviewer 2: ```suggestion
Operator secret used for authentication with Cloud Provider is stored in k8s API. 
```
For coherence

This means that actually etcd holds authentication credentials. 
Access to etcd or k8s-api are secured based on certificates validated in Vault, 
to which only personel in approriate Github GS organization have access to.   

***Reference docs***:  
[Github Vault auth](https://www.vaultproject.io/docs/auth/github.html)  

> [Vault ssh certificate](https://www.vaultproject.io/docs/secrets/ssh/signed-ssh-certificates.html)

Reviewer: ```suggestion
[Vault SSH certificate](https://www.vaultproject.io/docs/secrets/ssh/signed-ssh-certificates.html)
```
